scipy/stats/_multivariate.py

```diff
@@ -4275,3 +4277,448 @@ def rvs(self, size=1, random_state=None):
+        """
+        M, m, n, _, _, mncond = self._process_parameters(m, n)
+        M, n = M[..., np.newaxis], n[..., np.newaxis]
+        output = n * m/M * (M-m)/M * (M-n)/(M-1)
+        return self._checkresult(output, mncond, np.nan)
+
+    def cov(self, m, n):
+        """
+        Covariance matrix of the Multivariate Hypergeometric distribution.
+
+        Parameters
+        ----------
+        %(_doc_default_callparams)s
+
+        Returns
+        -------
+        cov : array_like
+            The covariance matrix of the distribution
+        """
+        # see [1]_ for the formula and [2]_ for implementation
+        # cov( x_i,x_j ) = -n * (M-n)/(M-1) * (K_i*K_j) / (M**2)
+        M, m, n, _, _, mncond = self._process_parameters(m, n)
+        M = M[..., np.newaxis, np.newaxis]
+        n = n[..., np.newaxis, np.newaxis]
+        output = (-n * (M-n)/(M-1) / (M**2) *
+                  np.einsum("...i,...j->...ij", m, m))
+        dim = m.shape[-1]
+        M, n = M[..., 0, 0], n[..., 0, 0]
+        # diagonal entries need to be computed differently
+        for i in range(dim):
+            output[..., i, i] = (n * (M-n)/(M-1) *
+                                 m[..., i]*(M-m[..., i]) / (M**2))
+
+        return self._checkresult(output, mncond, np.nan)
+
+    def rvs(self, m, n, size=1, random_state=None):
+        """
+        Draw random samples from a Multivariate Hypergeometric distribution.
+
+        Parameters
+        ----------
+        %(_doc_default_callparams)s
+        size : integer or iterable of integers, optional
+            Number of samples to draw (default 1).
+        %(_doc_random_state)s
+
+        Returns
+        -------
+        rvs : array_like
+            Random variates of shape (`size`, `len(p)`)
+
+        Notes
+        -----
+        %(_doc_callparams_note)s
+        """
+        M, m, n, _, _, mncond = self._process_parameters(m, n)
+
+        random_state = self._get_random_state(random_state)
+        # Generator is only available in numpy >= 1.17 while
+        # `multivariate_hypergeometric` distribution is only
+        # available in numpy >= 1.18
+        if (np.__version__ >= '1.18' and
+            isinstance(random_state, np.random.Generator)):  # noqa: E129
+            out = random_state.multivariate_hypergeometric(m, n, size)
+            return _squeeze_output(out)
+
+        if isinstance(size, int):
+            size = (size, )
+
+        rvs = np.empty(size + (m.shape[-1], ), dtype=m.dtype)
+        rem = M
+
+        # This sampler has been taken from numpy gh-13794
+        # https://github.com/numpy/numpy/pull/13794
+        for c in range(m.shape[-1] - 1):
+            rem = rem - m[..., c]
+            rvs[..., c] = ((n != 0) *
+                            random_state.hypergeometric(m[..., c], rem,
+                                                        n + (n == 0),
+                                                        size=size))
```

Comment:
Indentation nitpick (flake8 complains about this):
```suggestion
            rvs[..., c] = ((n != 0) *
                           random_state.hypergeometric(m[..., c], rem,
                                                       n + (n == 0),
                                                       size=size))
```

Reply:
done